### PR TITLE
Add benchmark folders, include benchmark_runner, xla_flags_libs and model libs for trillium.

### PR DESCRIPTION
**Pull request by suexu1025**
* Added maxstar model libs. xla flag libs.
* Added example llama benchmark test code.

### CODE REVIEW

File: benchmarks/llama3.1_v6e-256_benchmarks.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from maxtext_xpk_runner import HWConfig
+from maxtext_xpk_runner import SWconfig
+from maxtext_xpk_runner import xpk_benchmark_runner
+from maxtext_xpk_runner import XpkConfig
+
+
+DATE = '20241009'
+BASE_DOCKER_IMAGE = 'maxtext_base_image'
+
+ZONE = 'europe-west4'
```
Comment (Obliviour) [inline benchmarks/llama3.1_v6e-256_benchmarks.py:12]: Do you mind creating an argparser for the common arguments here: ZONE, PROJECT, CLUSTER_NAME, DEVICE_TYPE, NUM_SLICES so the user can plumb those in when running the script?

Comment (Obliviour) [reply benchmarks/llama3.1_v6e-256_benchmarks.py:12]: https://github.com/AI-Hypercomputer/xpk/blob/main/src/xpk/parser/common.py#L26 for reference

File: benchmarks/llama3.1_v6e-256_benchmarks.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+DEVICE_TYPE = 'v6e-256'
+NUM_SLICES = 1
+
+v6e_env_configs = SWconfig(
+    base_docker_image=BASE_DOCKER_IMAGE, libtpu_version=DATE
+)
+v6e_256_configs = HWConfig(num_slices=NUM_SLICES, device_type=DEVICE_TYPE)
+
+## todo: replace with llama3.1 model tests
+llama2_70b_4096 = BenchmarkRunner(
```
Comment (Obliviour) [inline benchmarks/llama3.1_v6e-256_benchmarks.py:24]: Nice, we can even use argparser to take in the model name argument. The benefit of this is we can have one common benchmark_runner.py file. Curious what you think of this approach?

**Review by Obliviour — COMMENTED** [review]
Thank you Qinwen! Let's work over chat to figure which models we want to put here first. I have some comments about how to make it easier to script the benchmark running, the format overall looks great!

File: benchmarks/xla_flags_library.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
+"""XLA Flags."""
```
Comment (Obliviour) [inline benchmarks/xla_flags_library.py:1]: We should go through this file and clean it up, I will help here

File: benchmarks/llama3.1_v6e-256_benchmarks.py (hunk 2 above)
Comment (suexu1025) [reply benchmarks/llama3.1_v6e-256_benchmarks.py]: Thanks for the suggestion, I think it's a good idea, pls find the updated code.

**Review by Obliviour — APPROVED** [review]
This generally LGTM using the custom arguments + one file for running this config is awesome. @gobbleturk can you please take a deeper look?

File: benchmarks/benchmark_runner.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
+import argparse
```
Comment (gobbleturk) [inline benchmarks/benchmark_runner.py:1]: Can we include comments at top of this file for what it is meant to do, example invocations, documentation for arguments?

File: benchmarks/maxtext_xpk_runner.py
Comment (gobbleturk) [inline benchmarks/maxtext_xpk_runner.py:1]: Can we also include comments at the top here - what is this file, example usage, arguments

File: benchmarks/maxtext_xpk_runner.py [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+    cluster_config: XpkConfig,
+    buffer_size: int,
+):
+  config_tuning_params = ''
+  for key, value in model.tuning_params.items():
+    config_tuning_params += f'{key}={value} '
+
+  install_libtpu_cmd = ''
+  if libtpu_type == LibTpuType.V6E_NIGHTLY_WHEEL:
+    install_libtpu_cmd += f' pip install *${libtpu_date}*.whl &&'
```
Comment (gobbleturk) [inline benchmarks/maxtext_xpk_runner.py:256]: Is this actually specific to v6e? Can this option be renamed to NIGHTLY_WHEEL if its not?

File: benchmarks/maxstar_model_configs.py [hunk 6]
```diff
@@ -0,0 +1,459 @@
+"""Shared Benchmark config for v6e orchestrations."""
```
Comment (gobbleturk) [inline benchmarks/maxstar_model_configs.py:1]: Can we have a more descriptive comment at the top.

If this file is meant to be v6e specific it should be renamed to maxtext_v6e_configs or similar. If not v6e specific we should change the comment to reflect this

Comment (gobbleturk) [reply benchmarks/maxstar_model_configs.py:1]: Generally I would prefer maxtext over maxstar, we can always change the name later if at some point add non-maxtext configs

**Review by gobbleturk — COMMENTED** [review]
This looks very cool, thank you! Ideally we should include an e2e test which exercises maxtext_xpk_runner, at least confirming it runs

File: benchmarks/benchmark_runner.py (hunk 4 above)
Comment (suexu1025) [reply benchmarks/benchmark_runner.py:28]: Added

File: benchmarks/maxtext_xpk_runner.py
Comment (suexu1025) [reply benchmarks/maxtext_xpk_runner.py:1]: Added top usage comments

File: benchmarks/maxtext_xpk_runner.py (hunk 5 above)
Comment (suexu1025) [reply benchmarks/maxtext_xpk_runner.py]: I think we can safely remove it, as now v6e libtpu can use nightly version using jax-release channel. @Obliviour to confirm?

File: benchmarks/maxstar_model_configs.py (hunk 6 above)
Comment (suexu1025) [reply benchmarks/maxstar_model_configs.py:1]: I think this is v6e specific tuned model configs. I will rename it ``` maxtext_v6e_model_configs ```. What do you think? @Obliviour

File: benchmarks/maxtext_xpk_runner.py [hunk 7]
```diff
@@ -0,0 +1,456 @@
...
+    num_slices=options.num_slices, device_type=options.device_type
+)
+
+model_runner = BenchmarkRunner(
+    model_name=benchmark_model,
+    software_config=v6e_env_configs,
+    hardware_config=v6e_256_configs,
+)
+
+xpk_benchmark_runner(cluster_config, [model_runner])
```
Comment (gobbleturk) [inline benchmarks/maxtext_xpk_runner.py:44]: Ah I understand files much better now thank you. Can you add that all of these steps are run for you in benchmarks/benchmark_runner - in fact you can just point to the documentation there
"This file contains data classes and runner logic to execute the XPK runs triggered by benchmarks/benchmark_runner.py"

**Review by gobbleturk — APPROVED** [review]
Thank you for this Qinwen and Victor, it looks very cool!

**Pull request merged**